### PR TITLE
paymentsservice: structure logs

### PR DESCRIPTION
**Pull request by naseemkullah**
Use Pino's [mergingObject](https://github.com/pinojs/pino/blob/master/docs/api.md#mergingobject-object) parameter to add context as JSON
rather than in the message parameter, which is now more generic (like an event name).

Also factor out logger so it is instantiated once for the whole service and upgrade pino.

Also remove unneeded comments.

### CODE REVIEW

File: src/paymentservice/index.js [hunk 1]
```diff
@@ -31,13 +28,13 @@ function chargeServiceHandler(call, callback) {
...
     })
-    logger.info(`PaymentService#Charge invoked by: ${JSON.stringify(call.request)}`)
+    logger.info({request: call.request}, "Charge request received.")
 
     const response = charge.charge(call.request)
     callback(null, response)
 
   } catch (err) {
-    logger.warn(err)
+    logger.warn({err})
```
Comment (naseemkullah) [inline src/paymentservice/index.js:37]: Pino applies their error serialization to Error objects named `err` out of the box 👍 

- https://github.com/pinojs/pino-std-serializers#exportserrerror
- https://getpino.io/#/docs/api?id=serializers-object

File: src/paymentservice/index.js [hunk 2]
```diff
@@ -63,9 +57,9 @@ server.addService(health.service, new health.Implementation({
 
 server.addService(hipsterShopPackage.hipstershop.PaymentService.service, { charge: chargeServiceHandler })
 
-server.bindAsync(`0.0.0.0:${process.env['PAYMENT_SERVICE_PORT']}`, grpc.ServerCredentials.createInsecure(), (err, port) => {
+server.bindAsync(`0.0.0.0:${process.env['PAYMENT_SERVICE_PORT'] ?? 50051}`, grpc.ServerCredentials.createInsecure(), (err, port) => {
```
Comment (puckpuck) [inline src/paymentservice/index.js:60]: I don't think we want to hardcode a default port for any of the services outside of the .env file.
```suggestion
server.bindAsync(`0.0.0.0:${process.env['PAYMENT_SERVICE_PORT']}`, grpc.ServerCredentials.createInsecure(), (err, port) => {
```

**Review by puckpuck — APPROVED** [review]
please remove the hardcoded default port, the rest looks great.

Thanks!

**Comment by naseemkullah** [comment]
> please remove the hardcoded default port, the rest looks great.
> 
> Thanks!

done, my pleasure

File: src/paymentservice/logger.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+const pino = require('pino');
+
+module.exports =  pino();
```
Comment (mic-max) [inline src/paymentservice/logger.js:3]:
```suggestion
module.exports = pino();
```

**Review by mic-max — APPROVED** [review]

**Comment by mic-max** [comment]
LGTM!

**Review by reyang — APPROVED** [review]

**Review by austinlparker — APPROVED** [review]

**Pull request merged**